acme/http.go

```diff
@@ -104,6 +104,22 @@ func retryAfter(v string) time.Duration {
 	return t.Sub(timeNow())
 }
 
+// responseChecker contains an isRetriableFn which reports if the request should be
+// retried based on the response code.
+type responseChecker struct {
+	// isRetriableFn reports whether a request can be retried based on the HTTP response status code.
+	// See Client.ShouldRetry doc comment.
+	isRetriableFn func(httpStatusCode int) bool
+}
+
+func (c *Client) responseChecker() *responseChecker {
+	f := c.ShouldRetry
+	if f == nil {
+		f = isRetriable
+	}
+	return &responseChecker{isRetriableFn: f}
+}
```

Comment:
Suggestion - just took a look at the `retryTimer()` method to see how it was structured; and came across this: 

https://github.com/golang/crypto/blob/4663e185863a1aee50d0486b326769f0bd22eb30/acme/http.go#L22-L23

```
// retryTimer encapsulates common logic for retrying unsuccessful requests.
// It is not safe for concurrent use.
```

Was thinking perhaps it might make sense to make the struct more generic (perhaps named `retries`), and then embed this new function there. 

So down below it would look like: 

```
c.retry.backoff()
c.retry.isRetriable(...)
```

Reply:
The benefit is that the retry logic doesn't span multiple structs

Reply:
good idea, let me noodle on this